src/Controllers/DatabaseController.js

```diff
@@ -18,7 +18,14 @@ function addWriteACL(query, acl) {
 function addReadACL(query, acl) {
   const newQuery = _.cloneDeep(query);
   //Can't be any existing '_rperm' query, we don't allow client queries on that, no need to $and
-  newQuery._rperm = { "$in" : [null, "*", ...acl]};
+  if (newQuery.hasOwnProperty('$or')) {
+    newQuery.$or = newQuery.$or.map(function(qobj) {
```

Comment:
It might be nice to add a comment here that provides a link to the MongoDB JIRA ticket and a note that we are intentionally distributing the _rperm predicate.